src/geometamaker/models.py

```
@@ -132,7 +132,130 @@ def __post_init__(self):
+
+        """
+        if self.contact is None:
+            self.contact = ContactSchema()
+        if organization is not None:
+            self.contact.organization = organization
+        if individual_name is not None:
+            self.contact.individual_name = individual_name
+        if position_name is not None:
+            self.contact.position_name = position_name
+        if email is not None:
+            self.contact.email = email
+
+    def get_contact(self):
+        """Get metadata from a contact section.
+
+        Returns:
+            ContactSchema
+
+        """
+        return self.contact
+
+    def set_license(self, title=None, path=None):
+        """Add a license for the dataset.
+
+        Either or both title and path are required if there is a license.
+        Call with no arguments to remove license info.
+
+        Args:
+            title (str): human-readable title of the license
+            path (str): url for the license
+
+        """
+        if self.license is None:
+            self.license = LicenseSchema()
+        license_dict = {}
+        license_dict['title'] = title if title else ''
+        license_dict['path'] = path if path else ''
+
+        # TODO: DataPackage/Resource allows for a list of licenses.
+        # So far we only support one license per resource.
+        self.license = LicenseSchema(**license_dict)
+
+    def get_license(self):
+        """Get ``license`` for the dataset.
+
+        Returns:
+            models.LicenseSchema
+
+        """
+        # TODO: DataPackage/Resource allows for a list of licenses.
+        # So far we only support one license per resource.
+        return self.license
+
+    def replace(self, other):
+        if isinstance(other, _Metadata):
+            return dataclasses.replace(
+                self, **{k: v for k, v in other.__dict__.items() if v is not None})
+        return NotImplementedError
```

Wouldn't this need to be `raise`d instead of `return`ed?

---

woops, thanks!

---

I just noticed this still wasn't right. I think it's more helpful to raise a `TypeError` here with a helpful message. I also renamed `_Metadata` to `BaseMetadata`. I don't intend it to be a user-facing class, but the `Base` prefix seems a bit more conventional.